### ISSUE TITLE
Relative url containing only a fragment also affects the path

### ISSUE THREAD

**Issue by hadley** [jeroen/curl#366]
``` r
curl::curl_parse_url("#foo", "http://x.com/a/b/c")$path
#> [1] "/a/b/"
```

<sup>Created on 2024-12-23 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

IMO this should be `/a/b/c`.

**Comment by jeroen** [comment]
Yikes. Looks like this was actually fixed already in recent libcurl (I think 8.8.0) but that doesn't help us much. I'll add a workaround I guess.

**Comment by jeroen** [comment]
Checking with upstream curl if this is expected: https://github.com/curl/curl/issues/15836
